src/model_repository_manager.cc

```diff
@@ -256,6 +256,11 @@ GetModifiedTime(const std::string& path)
   }
 
   for (const auto& child : contents) {
+    if (child.empty()) {
```

Comment:
Why does a misrepresented file as a directory would have an empty string as a content?
I would have assumed that contents to be an empty set. 
Usually, an actual directory will hold "." and "..".
I am concerned if we are missing some important cases.

Reply:
Can there be cases if a directory contains a file with name "" ?

Reply:
> Why does a misrepresented file as a directory would have an empty string as a content? I would have assumed that contents to be an empty set.

I think for most cloud storage systems, there is no directory structures which each "file" is stored with its full path "/dir1/dir2/file" (i.e. on a flat table). If for any reason "/dir1/dir2/file" is misinterpreted as a directory, then I imagine any resources nested within "/dir1/dir2/file/" would be "" and it would be invalid, because "/dir1/dir2/file/" is simply "/dir1/dir2/file" which is somewhat an empty set.

> Usually, an actual directory will hold "." and "..".

Yes on Linux, but I think the layout on cloud storages can be different.

> Can there be cases if a directory contains a file with name "" ?

I think it is invalid, because if a parent directory contains a child file with name "", then the child file will have a full path of "/dir1/dir2/", which is equivalent to a full path of "/dir1/dir2" that matches the full path of the parent directory. A full path must be deterministic which cannot reference both the parent directory and the child file at the same time, so it leads to a contradiction. Thus, the child file cannot have name "".

I found some discussions online that some [file systems might allow empty file name](https://stackoverflow.com/questions/68531715/can-an-empty-string-be-a-legal-file-name), but I think it would be invalid on Triton because we use the "/dir1/dir2/file" structure that will leads to the above contradiction if a file name is empty.